### PR TITLE
adding currency change and presets change to event system

### PR DESCRIPTION
**Pull request by ScottDowne**
This is a WIP

### CODE REVIEW

**Comment by ScottDowne** [comment]
No longer a WIP, it's not a shiny new patch. @alicoding r?

File: components/amount-buttons.jsx [hunk 1]
```diff
@@ -16,17 +16,17 @@ var AmountButton = React.createClass({
...
         <label htmlFor={"amount-" + this.props.value} className="amount-button large-label-size">
+          { this.props.currencyCode && this.props.value ?
           <FormattedNumber
             minimumFractionDigits={0}
             value={this.props.value}
             style="currency"
-            currency={this.props.currencyCode || "usd"}
-          />
+            currency={this.props.currencyCode}
+          /> : <span>&nbsp;</span> }
```
Comment (alicoding) [inline components/amount-buttons.jsx]: why you need a whitespace?

File: components/amount-buttons.jsx [hunk 2]
```diff
@@ -64,7 +64,12 @@ var AmountOtherButton = React.createClass({
             value={this.props.amount}
           />
           <label htmlFor="amount-other" className="large-label-size">
-            <span className="currency-symbol-container">{this.props.currencySymbol}</span>
+            <span className="currency-symbol-container">
+              { this.props.currencySymbol ?
+                <span>{this.props.currencySymbol}</span> :
+                <span>&nbsp;</span>
```
Comment (alicoding) [inline components/amount-buttons.jsx]: and here...

File: components/amount-buttons.jsx (hunk 1 above)
Comment (ScottDowne) [inline components/amount-buttons.jsx]: So if the page is still parsing for the currency/amount/presets, it doesn't display incorrect info or a flat buttons with no height.

Instead, I display nothing, and add in the space so the button has the correct height.

Comment (alicoding) [inline components/amount-buttons.jsx]: Just want to make sure you know you can just do `''` but if you intentionally want a space then this is fine.

File: components/donate-button.jsx [hunk 3]
```diff
@@ -25,9 +25,19 @@ module.exports = React.createClass({
...
+    if (state === "currency") {
+      this.setState({
+        currency: value
+      });
+    }
+  },
   mixins: [IntlMixin],
   render: function() {
-    if (this.state.amount) {
+    if (this.state.amount ) {
```
Comment (alicoding) [inline components/donate-button.jsx]: remove this whitespace :P

File: components/amount-buttons.jsx (hunk 1 above)
Comment (ScottDowne) [inline components/amount-buttons.jsx]: the '' creates a flat button. Need a nbsp to force it to have something.

**Comment by alicoding** [comment]
r+

File: components/amount-buttons.jsx (hunk 1 above)
Comment (ScottDowne) [inline components/amount-buttons.jsx]: I expected the '' to do that too, then I tried ' ' which I was also annoyed didn't work. Then I tried '&nbsp;' which displayed the text. lol, so adding it to a span not as text fixed it.

**Pull request merged**